pkg/R/RDD.R

```diff
@@ -1348,16 +1363,16 @@ setMethod("unionRDD",
           function(x, y) {
             if (x@env$serialized == y@env$serialized) {
               jrdd <- callJMethod(getJRDD(x), "union", getJRDD(y))
-              union.rdd <- RDD(jrdd, x@env$serialized)
+              union.rdd <- RDD(jrdd, x@env$serialized, colNames = x@colNames)
             } else {
               # One of the RDDs is not serialized, we need to serialize it first.
-              if (!x@env$serialized) {
+              if (x@env$serialized == "string") {
```

Comment:
@cafreeman looking at the [newly added test](https://github.com/cafreeman/SparkR-pkg/commit/8c3b8c5e022e7f747188613bebf5c2b2faedb939), it seems to me the test doesn't hit this else branch, since the two RRDD's serialization modes are equal (RowSerialized).